src/pkg/projects/Microsoft.NETCore.App/src/Microsoft.NETCore.App.depproj

```diff
@@ -74,6 +74,44 @@
     <Error Condition="!Exists('$(%(_requiredProperty.Identity))')" Text="Required property %(_requiredProperty.Identity) with value '$(%(_requiredProperty.Identity))' does not exist." />
   </Target>
 
+  <Target Name="GenerateHashVersionsFile" DependsOnTargets="GetPackagePaths;ResolveNuGetPackages">
+    <PropertyGroup>
+      <CoreFxVersionFile 
+        Condition="$([System.String]::new('%(Reference.Identity)').ToLowerInvariant().Contains('system.runtime.dll'))">
+          $(PackagesDir)%(Reference.NuGetPackageId)/%(Reference.NuGetPackageVersion)/version.txt
+      </CoreFxVersionFile>
+      <CoreClrVersionFile Condition="'$(NuGetRuntimeIdentifier)' != ''">
+          $(_runtimePackagePath)/version.txt
+      </CoreClrVersionFile>
+    </PropertyGroup>
+    <ReadLinesFromFile File="$(CoreFxVersionFile)" Condition="Exists('$(CoreFxVersionFile)')">
+      <Output TaskParameter="Lines" PropertyName="CoreFxHash" />
+    </ReadLinesFromFile>
+    <ReadLinesFromFile File="$(CoreClrVersionFile)" Condition="Exists('$(CoreClrVersionFile)')">
+      <Output TaskParameter="Lines" PropertyName="CoreClrHash" />
+    </ReadLinesFromFile>
+
+    <Error Condition="!Exists('$(CoreFxVersionFile)')" Text="Failed to find the corefx version.txt file at $(CoreFxVersionFile)" />
+
+    <ItemGroup>
+      <_VersionsFileLines Include="core-setup $(LatestCommit)" />
+      <_VersionsFileLines Include="corefx $(CoreFxHash)" />
+      <_VersionsFileLines Include="dotnet/versions/corefx $(CoreFxCurrentRef)" />
+      <_VersionsFileLines Include="coreclr $(CoreClrHash)" Condition="'$(CoreClrHash)'!=''" />
```

Comment:
As discussed offline, instead of a ToDo I'll just log an issue about fixing this once we move to packageReference, as well as add one more hash for standard repo.